src/lib.rs

```diff
@@ -69,11 +69,34 @@ impl Order {
         }
     }
 
+    /// Generate a Certificate Signing Request for the given names and request finalization
+    ///
+    /// Uses the rcgen crate to generate a Certificate Signing Request (CSR) for the given
+    /// Subject Alternative Names, then calls [`Order::finalize()`] with it. Returns the generated
+    /// private key, serialized as PEM.
+    ///
+    /// After this succeeds, call [`Order::certificate()`] to retrieve the certificate chain once
+    /// the order is in the appropriate state.
+    #[cfg(feature = "rcgen")]
+    pub async fn finalize_for(&mut self, names: Vec<String>) -> Result<String, Error> {
+        use rcgen::{CertificateParams, DistinguishedName, KeyPair};
+
+        let mut params = CertificateParams::new(names)?;
+        params.distinguished_name = DistinguishedName::new();
```

Comment:
unrelated: I'd love to kill this step upstream.... IMO for EE certs in 2025 not having a common name should be the default.